pkg/semver/semver_test.go

```diff
@@ -1,17 +1,234 @@
 package semver
 
 import (
-	"reflect"
+	"errors"
+	"fmt"
 	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	version_1_2_3 = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 3,
+	}
+	version_1_2_2 = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 2,
+	}
+	version_1_2_3_WithPR = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 3,
+		Pre: []PRVersion{
+			{
+				VersionNum: 4,
+				IsNum:      true,
+			},
+			{
+				VersionStr: "alpha5",
+			},
+		},
+	}
+	version_1_2_3_WithBuild = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 3,
+		Build: []string{
+			"6",
+			"alpha7",
+		},
+	}
+	version_1_2_3_WithPRAndBuild = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 3,
+		Pre: []PRVersion{
+			{
+				VersionNum: 4,
+				IsNum:      true,
+			},
+			{
+				VersionStr: "alpha5",
+			},
+		},
+		Build: []string{
+			"6",
+			"alpha7",
+		},
+	}
+	version_1_2_2_WithPRAndBuild = Version{
+		Major: 1,
+		Minor: 2,
+		Patch: 2,
+		Pre: []PRVersion{
+			{
+				VersionNum: 4,
+				IsNum:      true,
+			},
+			{
+				VersionStr: "alpha5",
+			},
+		},
+		Build: []string{
+			"6",
+			"alpha7",
+		},
+	}
 )
 
+func TestOperators(t *testing.T) {
+	tests := []struct {
+		name    string
+		op      func(v Version, o Version) bool
+		version Version
+		other   Version
+		want    bool
+	}{
+		// Equal
+		{
+			name:    "Equals: same version should be equal",
+			op:      Version.Equals,
+			version: version_1_2_3,
+			other:   version_1_2_3,
+			want:    true,
+		},
+		{
+			name:    "Equals: same version should be equal - with prerelease and build",
+			op:      Version.Equals,
+			version: version_1_2_3_WithPRAndBuild,
+			other:   version_1_2_3_WithPRAndBuild,
+			want:    true,
+		},
+		{
+			name:    "Equals: different version should be unequal",
+			op:      Version.Equals,
+			version: version_1_2_3,
+			other:   version_1_2_2,
+			want:    false,
+		},
+		{
+			name:    "Equals: different prerelease should not be equal",
+			op:      Version.Equals,
+			version: version_1_2_3,
+			other:   version_1_2_3_WithPR,
+			want:    false,
+		},
+
+		// NE
```

Comment:
Essentially all Equal tests, but negated?

Reply:
Yes, but with `Version.NE` to cover that function.